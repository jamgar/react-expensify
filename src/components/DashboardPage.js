import React from 'react'
import { Link } from 'react-router-dom'

const DashboardPage = () => (
  <div className="content-container">
    <p>
      <Link to='/expense-dashboard' className="button">View Expenses</Link>
    </p>
    <p>
      <Link to='/income-dashboard' className="button">View Income</Link>
    </p>
  </div>
)

export default DashboardPage
