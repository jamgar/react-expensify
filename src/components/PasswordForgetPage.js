import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { history } from '../routers/AppRouter'
import { startPasswordReset } from '../actions/auth'

export class PasswordForgetPage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      error: ''
    }
  }
  isInvalid = () => {
    this.state.email === ''
  }
  onEmailChange = (e) => {
    const email = e.target.value
    this.setState({ email })
  }
  onSubmit = (e) => {
    e.preventDefault()
    if (this.isInvalid()) {
      this.setState({ error: 'Please enter an email'})
    } else {
      this.setState({ error: '' })
      this.props.startPasswordReset(this.state.email)
        .then(() => {
          console.log('Email sent...')
          this.setState({ error: 'Please check your email for a password reset link.'})
          setTimeout(() => {
            history.push('/')
          }, 5000)
        })
        .catch(error => {
          let message

          switch (error.code) {
            case 'auth/user-not-found':
              message = 'User with this email not found.'
              break;
            case 'auth/invalid-email':
              message = 'The email is invalid.'
              break;
            default:
              message = 'Something went wrong.'
              console.log('Auth Error:', error)
          }
          this.setState({ error: `${message} Please try again.`})
        })
    }
  }
  render() {
    const {
      email,
      error
    } = this.state

    return (
      <div>
        <div className="page-header page-header--centered">
          <div className="content-container">
            <h1 className="page-header__title">Password Forget</h1>
          </div>
        </div>
        <div className="content-container content-container--sm">
          <form className="form" onSubmit={this.onSubmit}>
            {error && <p className="form__error">{error}</p>}
            <input
              type="email"
              className="text-input"
              placeholder="Email Address"
              autoFocus
              value={email}
              onChange={this.onEmailChange}
            />
            <button className="button" type="submit">
              Send Reset Email
            </button>
            <Link className="link" to='/'>
              Cancel
            </Link>
          </form>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startPasswordReset: (email) => dispatch(startPasswordReset(email))
})

export default connect(undefined, mapDispatchToProps)(PasswordForgetPage)
