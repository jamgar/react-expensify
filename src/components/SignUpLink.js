import React from 'react'
import { Link } from 'react-router-dom'

export const SignUpLink = () => (
  <Link className="link" to='/signup'>
    Sign Up
  </Link>
)
