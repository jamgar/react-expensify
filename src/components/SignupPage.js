import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { startCreateUserWithEmailAndPassword } from '../actions/auth'

export class SignupPage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      passwordConfirmation: '',
      error: ''
    }
  }
  isInvalid = () => (
    this.state.password !== this.state.passwordConfirmation ||
    this.state.password === '' ||
    this.state.email === ''
  )
  onEmailChange = (e) => {
    const email = e.target.value
    this.setState({ email })
  }
  onPasswordChange = (e) => {
    const password = e.target.value
    this.setState({ password })
  }
  onPasswordConfirmationChange = (e) => {
    const passwordConfirmation = e.target.value
    this.setState({ passwordConfirmation })
  }
  onSubmit = (e) => {
    e.preventDefault()
    const { email, password } = this.state
    if (this.isInvalid()) {
      this.setState({ error: 'Invalid email or passwords do not match.'})
    } else {
      this.setState({ error: '' })
      this.props.startCreateUserWithEmailAndPassword(email, password)
    }
  }
  render() {
    const {
      email,
      password,
      passwordConfirmation,
      error
    } = this.state
    return (
      <div>
      <div className="page-header page-header--centered">
        <div className="content-container">
          <h1 className="page-header__title">Sign Up</h1>
        </div>
      </div>
      <div className="content-container content-container--sm">
        <form className="form" onSubmit={this.onSubmit}>
          {error && <p className="form__error">{error}</p>}
          <input
            type="email"
            className="text-input"
            placeholder="Email Address"
            autoFocus
            value={email}
            onChange={this.onEmailChange}
          />
          <input
            type="password"
            className="text-input"
            placeholder="Password"
            value={password}
            onChange={this.onPasswordChange}
          />
          <input
            type="password"
            className="text-input"
            placeholder="Password Confirmation"
            value={passwordConfirmation}
            onChange={this.onPasswordConfirmationChange}
          />
          <button type="submit" className="button">
            Sign Up
          </button>
          <p>
            <Link className="link" to='/'>
              Cancel
            </Link>
          </p>
        </form>
      </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startCreateUserWithEmailAndPassword: (email, password) => dispatch(startCreateUserWithEmailAndPassword(email, password))
})

export default connect(undefined, mapDispatchToProps)(SignupPage)

export const SignUpLink = () => (
  <p>
    <Link className="link" to='/signup'>
      Sign Up
    </Link>
  </p>
)
