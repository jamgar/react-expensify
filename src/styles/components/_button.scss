.button {
  background: $bright-blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  transition: .5s;
}
.button:hover {
  background: darken($bright-blue, 5%);
}
.button--link,
.button--link:hover {
  background: none;
}
.button--secondary {
  background-color: #888888;
}
