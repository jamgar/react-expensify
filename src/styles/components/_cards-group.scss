.cards-group {
  display: flex;
}
.card {
  border: 1px solid darken($off-white, 7%);
  margin: $s-size;
  width: 50%;
}
.card__header {
  padding: $s-size;
}
.card__header--dark {
  background: $dark-blue;
  color: white;
}
.card__title {
  margin: 0;
  padding: 0;
  text-align: center;
}
.card__content {
  font-size: $font-size-large;
  margin: $s-size;
  padding: $s-size;
}
.card__footer {
  padding: $s-size;
  text-align: center;
}
