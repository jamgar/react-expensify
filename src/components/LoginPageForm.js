import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { SignUpLink } from './SignUpLink'
import { startSignInWithEmailAndPassword } from '../actions/auth'

class LoginPageForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      error: ''
    }
  }
  isInvalid = () => (
    this.state.email === '' || this.state.password === ''
  )
  onEmailChange = (e) => {
    const email = e.target.value
    this.setState({ email })
  }
  onPasswordChange = (e) => {
    const password = e.target.value
    this.setState({ password })
  }
  onSubmit = (e) => {
    const { email, password } = this.state
    e.preventDefault()
    if (this.isInvalid()) {
      this.setState({ error: 'Invalid email or password.'})
    } else {
      this.setState({ error: '' })
      this.props.startSignInWithEmailAndPassword(email, password)
    }
  }
  render() {
    const {
      email,
      password,
      error
    } = this.state
    return (
      <div>
        <div className="page-header page-header--centered">
          <div className="content-container">
            <h1 className="page-header__title">Login</h1>
          </div>
        </div>
        <div className="content-container content-container--sm">
          <form className="form" onSubmit={this.onSubmit}>
            {error && <p className="form__error">{error}</p>}
            <input
              type="email"
              className="text-input"
              placeholder="Email Address"
              autoFocus
              value={email}
              onChange={this.onEmailChange}
            />
            <input
              type="password"
              className="text-input"
              placeholder="Password"
              value={password}
              onChange={this.onPasswordChange}
            />
            <button type="submit" className="button">
              Log In
            </button>
            <Link className="link" to='/'>
            Cancel
            </Link>
          </form>
          <SignUpLink />
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startSignInWithEmailAndPassword: (email, password) => dispatch(startSignInWithEmailAndPassword(email, password))
})

export default connect(undefined, mapDispatchToProps)(LoginPageForm)
