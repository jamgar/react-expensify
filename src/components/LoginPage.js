import React from 'react'
import { connect } from 'react-redux'
import { startLoginWithGoogle } from '../actions/auth'
import { SignUpLink } from './SignupPage'

export const LoginPage = ({ startLoginWithGoogle }) => (
    <div className="box-layout">
      <div className='box-layout__box'>
        <h1 className="box-layout__title">Expensify</h1>
        <p>Time to take charge of you expenses.</p>
        <button className="button" onClick={startLoginWithGoogle}>Login with Google</button>
        <SignUpLink />
      </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLoginWithGoogle: () => dispatch(startLoginWithGoogle())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)
