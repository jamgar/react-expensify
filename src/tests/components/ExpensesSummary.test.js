import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should render ExpensesSummary with empty expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={234} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={23400} />)
  expect(wrapper).toMatchSnapshot()
})
