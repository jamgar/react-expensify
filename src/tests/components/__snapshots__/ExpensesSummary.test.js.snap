// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
     
    expenses
     totalling 
    $2.34
  </h2>
</div>
`;

exports[`should render ExpensesSummary with empty expenses 1`] = `
<div>
  <h2>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h2>
</div>
`;

exports[`should render ExpensesSummary with expenses 1`] = `
<div>
  <h2>
    Viewing 
     
    expenses
     totalling 
    $234.00
  </h2>
</div>
`;
